sdk/servicebus/azure-servicebus/azure/servicebus/_common/receiver_mixins.py

```diff
@@ -71,6 +83,25 @@ def _on_attach(self, source, target, properties, error):
     def _populate_message_properties(self, message):
         pass
 
+    @contextmanager
+    def _receive_trace_context_manager(self, message=None, span_name=SPAN_NAME_RECEIVE):
+        # type: (Optional[Union[Message, Iterable[Message]]], str) -> Iterator[None]
+        """Tracing"""
+        span_impl_type = settings.tracing_implementation()  # type: Type[AbstractSpan]
+        if span_impl_type is None:
+            yield
+        else:
+            receive_span = span_impl_type(name=span_name)
+            self._add_span_request_attributes(receive_span)  # type: ignore  # pylint: disable=protected-access
+            receive_span.kind = SpanKind.CONSUMER
+
+            # If it is desired to create link before span open
+            if message:
+                trace_link_message(message, receive_span)
+
+            with receive_span:
+                yield
```

Comment:
shall we move it to utils.py for symmetry with send_trace_context_manager?

`_receive_trace_context_manager` would call `self._add_span_request_attributes(receive_span)` while in `send_trace_context_manager` we don't. Is it the reason why `_receive_trace_context_manager` has to be on the receiver class?

Reply:
Your understanding is correct (re: the motivation).  Could still pass the receiver in if logical locality seems more sensible, I'll admit I was torn along roughly the same lines.  I'll sleep on it and likely merge them into utils, since it's very special-purpose rather than class-mental-model-oriented.

Reply:
(have merged into utils++)